neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/impl/BoltNeo4jConnectionImpl.java

```diff
@@ -277,31 +276,30 @@ private void setBookmark() throws SQLClientInfoException {
 	@Override public void close() throws SQLException {
 		try {
 			if (!this.isClosed()) {
-				if (this.transaction != null) {
-					this.transaction.close();
-					this.transaction = null;
-				}
-                if (this.session != null) {
-                    this.session.close();
-                    this.session = null;
-                    this.driver.close();
-                    this.driver = null;
-                }
-            }
+				closeTransaction();
+				closeSession(true);
```

Comment:
why does this close the driver?

Reply:
Because close the `Connection` instance, so makes sense here close everything

Reply:
No the driver should not be closed as it is used and cached across connections. Otherwise we would recreate drivers for each connection which would be super expensive and wasteful.

Reply:
Done